backend/cmd/headlamp.go

```diff
@@ -307,14 +308,20 @@ func StartHeadlampServer(config *HeadlampConfig) {
 			RedirectURL:  getOidcCallbackURL(r, config),
 			Scopes:       append([]string{oidc.ScopeOpenID}, oidcAuthConfig.Scopes...),
 		}
-
-		oauthRequestMap[cluster] = &OauthConfig{Config: oauthConfig, Verifier: verifier, Ctx: ctx}
-
-		http.Redirect(w, r, oauthConfig.AuthCodeURL(cluster), http.StatusFound)
+		/* we encode the cluster to base64 and set it as state so that when getting redirected
+		by oidc we can use this state value to get cluster name
+		*/
+		state := base64.StdEncoding.EncodeToString([]byte(cluster))
+		oauthRequestMap[state] = &OauthConfig{Config: oauthConfig, Verifier: verifier, Ctx: ctx}
+		http.Redirect(w, r, oauthConfig.AuthCodeURL(state), http.StatusFound)
 	}).Queries("cluster", "{cluster}")
 
 	r.HandleFunc("/oidc-callback", func(w http.ResponseWriter, r *http.Request) {
 		state := r.URL.Query().Get("state")
+		decodedState, err := base64.StdEncoding.DecodeString(state)
+		if err != nil {
```

Comment:
We should probably refer this error in the HTTP error and/or log it, so eventual bug reports are more complete.

Reply:
fixed